### PR TITLE
Remove on-import messages; change others to warnings

### LINKED ISSUE #346: Stdout flooded by the welcome message
When using spacepy to open a CDF file and stream the content to stdout, the welcome message appears in the stream (leading to app crashes downstream).

For the moment, I use the following code as a work around:

```python
try:
    # keep track of the previous stdout value
    stdout_ = sys.stdout
    stream = io.StringIO()
    sys.stdout = stream
    from spacepy.pycdf import CDF
    # restore the previous stdout
    sys.stdout = stdout_
except Exception as exc:
    raise Exception("Error during spacepy import") from exc
```

But it is a bit tedious. Do you have another solution? Is this message really useful ?

Note: the message disappears once the .spacepy folder is created which makes the error somewhat unpredictable.

### PR DESCRIPTION
**Pull request by jtniehof**
This PR removes several messages that show up on SpacePy import and changes a few others to warnings. Closes #346.

- Docs are updated to make information about `DOT_FLN` and the `.spacepy` directory more obvious so this information is apparent despite the loss of the first-import message (ac9db898ba1a1bfd9d88d1091c355ea09d648cb7)
- The no OMNI data message is now a warning (trappable) instead of a print statement (a3e41372a1d513354c82908ec084fe351baa6d60). In order to make this less of a mess, this no longer distinguishes between "no HDF5 OMNI data" and "no OMNI data at all" (546a7c01fb6ed6c9985830b45226589fb60fcc20); we stopped using the pickle in 2013 (195704003a775e59726e09a675726e2330eeb7f4).
- To compensate for not having a general `update` message on first import, `time` now checks if the leapsecond file is out of date (defined as "is it possible there was a leapsecond between the last one in the file and now?") (f25c755b21157fd741825d5078c8002919539c23) This requires preserving the mtime on `tai-utc.dat` when first imported (9718f31117c5a144da6acc0c55574d5bc566d9cc). I made this warning configurable (6c50667714c3305eed2f4e48c683ae0b591c8410) but honestly I wonder if that's overkill.
- Finally, with all these other messages in place, the first-import print statement is axed (77062488dab504a2162af083186141d078bf7e1c)

## PR Checklist

- [X] Pull request has descriptive title
- [X] Pull request gives overview of changes
- [X] New code has inline comments where necessary
- [X] (N/A) Any new modules, functions or classes have docstrings consistent with SpacePy style
- [X] Added an entry to CHANGELOG if fixing a major bug or providing a major new feature
- [X] (N/A) New features and bug fixes should have unit tests
- [X] Relevant issues are linked to (e.g. `See issue #` or `Closes #`)

The new functionality in here is actually pretty minor, and making unit tests seems more effort than it's worth (it would involve doing a bunch of setting environment variables, making temporary directories, and `reload`ing SpacePy over and over). So I decided to skip unit tests for these warnings.

### CODE REVIEW

File: spacepy/omni.py
Comment (balarsen) [inline spacepy/omni.py:417]: I like this solution a lot more!

**Review by balarsen — APPROVED** [review]
@jtniehof this all looks good and should address the original concern. As a related aside I have been using GNU parallel a lot lately and it has a "cite me" message to the term and it is annoying.

**Comment by balarsen** [comment]
@jtniehof you are done here? I will merge if so.

**Comment by jtniehof** [comment]
I'm good. @spacecataz chimed in on #346 so he might want to take a look at this before merge. Give him a day or so if he doesn't comment?

**Review by spacecataz — APPROVED** [review]

File: spacepy/time.py
Comment (balarsen) [inline spacepy/time.py]: It may pay to just be less smart here. `goodthrough = lastknown + datetime.timedelta(months=7)` is likely good enough to catch really out of date systems.

**Pull request merged**
